src/docs/components/styling.md

```diff
@@ -5,165 +5,207 @@ url: /docs/styling
+CSS custom properties, also often referred to as CSS variables, are used to contain values that can then be used in multiple CSS declarations. For example, we can create a custom property called `--color-primary` and assign it a value of `blue`.
 
-[CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables) are a lot like [Sass Variables](https://ionicframework.com/docs/theming/sass-variables/), but built into the browser. CSS Variables allow you to specify CSS properties that can be used across your app.
+```css
+:host {
+  --color-primary: blue;
+}
+```
 
-### Use Case
+And then we can use that custom property to style different parts of our component
 
-One use case for CSS Variables is colors. If your app has a primary brand color that is used across your app then instead of writing that same color out each place you need it in your app you can create a variable for it and then use that variable anywhere you need that color in your app. Also, if you ever need to change this color you will only have to change the variable and then it will be updated across your app.
+```css
+h1 {
+  color: var(--color-primary);
+}
+```
 
-### Using CSS Variables in Stencil
+### Customizing Components with Custom Properties
 
-Here are the recommended steps to use CSS Variables in Stencil:
+CSS custom properties can allow users of a component to customize a component’s styles from the light DOM. Consider a `shadow-card` component that uses a custom property for the color of the card heading.
 
-- Create a CSS file to hold your variable definitions. We normally recommend creating a `variables.css` file in `src/global/`
-- You can then put this config `globalStyle: 'src/global/variables.css'` into your `stencil.config.ts` file.
+```css
+:host {
+  --heading-color: black;
+}
 
-That's it! Now you can start defining your variables.
+.heading {
+  color: var(--heading-color);
+}
+```
 
-### Defining CSS Variables
+> **NOTE**: CSS custom properties must be declared on the `Host` element (`:host`) in order for them to be exposed to the consuming application.
 
-Here is an example of defining a CSS Variable:
+The `shadow-card` heading will have a default color of `black`, but this can now be changed in the light DOM by selecting the `shadow-card` and changing the value of the `--heading-color` custom property.
 
 ```css
-/* inside our src/global/variables.css file */
+shadow-card {
+  --heading-color: blue;
+}
+```
 
-:root {
-  --app-primary-color: #488aff;
+## CSS Parts
+
+CSS custom properties can be helpful for customizing components from the light DOM, but they are still a little limiting as they only allow a user to modify specific properties. For situations where users require a higher degree of flexibility, we recommend using the [CSS `::part()` pseudo-element](https://developer.mozilla.org/en-US/docs/Web/CSS/::part). You can define parts on elements of your component with the “part” attribute.
+
+```tsx
+export class ShadowCard {
```

Comment:
To make this example a little more "plug n play" (via copy/paste) can we add the `@Component` decorator to it?